lib/debug_socket.rb

```diff
@@ -16,7 +17,10 @@ def self.logger
   end
 
   def self.start(path)
-    raise "debug socket thread already running" if @thread
+    pid = Process.pid
+    raise "debug socket thread already running for this process" if @thread && @pid == pid
```

Comment:
I guess I mean, the same fork can open infinite debug sockets. Because `@pid == pid` will always be false.